src/Sulu/Component/Content/Document/Subscriber/OrderSubscriber.php

```diff
@@ -46,52 +35,45 @@ public static function getSubscribedEvents()
     {
         return array(
             Events::PERSIST => 'handlePersist',
-            Events::HYDRATE => 'handleHydrate',
+            Events::METADATA_LOAD => 'handleMetadataLoad',
         );
     }
 
+    public function handleMetadataLoad(MetadataLoadEvent $event)
+    {
+        $metadata = $event->getMetadata();
+
+        if (false === $metadata->getReflectionClass()->isSubclassOf(OrderBehavior::class)) {
+            return;
+        }
+
+        $metadata->addFieldMapping('suluOrder', array(
+            'encoding' => 'system',
+            'property' => self::FIELD,
+        ));
+    }
+
     /**
      * @param PersistEvent $event
      */
     public function handlePersist(PersistEvent $event)
     {
-        $node = $event->getNode();
         $document = $event->getDocument();
 
-        if (false == $this->supports($document)) {
+        if (false == $document instanceof OrderBehavior) {
             return;
         }
 
-        $propertyName = $this->encoder->systemName(self::FIELD);
-
-        if ($node->hasProperty($propertyName)) {
+        // TODO: This does not seem quite right..
+        if ($document->getSuluOrder()) {
             return;
         }
 
+        $node = $event->getNode();
         $parent = $node->getParent();
         $nodeCount = count($parent->getNodes());
         $order = ($nodeCount + 1) * 10;
```

Comment:
What exactly should this subscriber do? But you are right, it also feels weird to me.